src/request/sign-transaction/LayoutStandard.js

```diff
@@ -1,71 +1,88 @@
+        if (request.shopLogoUrl) {
+            const $shopLogo = document.createElement('img');
+            $shopLogo.src = request.shopLogoUrl.href;
+            $recipientIdenticon.classList.add('clip');
+            $recipientIdenticon.appendChild($shopLogo);
+            $shopLogo.addEventListener('error', () => {
+                $shopLogo.remove();
+                $recipientIdenticon.classList.remove('clip');
+                // eslint-disable-next-line no-new
+                new Identicon(request.transaction.recipient.toUserFriendlyAddress(), $recipientIdenticon);
+            });
+        } else {
+            // eslint-disable-next-line no-new
+            new Identicon(request.transaction.recipient.toUserFriendlyAddress(), $recipientIdenticon);
+        }
 
-                        <i class="arrow nq-icon chevron-right"></i>
+        const $recipientAddresses = ($recipient.querySelectorAll('.address > .chunk'));
+        /** @type {string[]} */
+        const recipientAddressChunks = (
+            request.transaction.recipient
+                .toUserFriendlyAddress()
+                .replace(/[+ ]/g, '').match(/.{4}/g)
+        );
+        $recipientAddresses.forEach(($element, x) => {
+            $element.textContent = recipientAddressChunks[x];
+        });
 
-                        <div class="account">
-                            <div class="identicon" id="recipient-identicon"></div>
-                            <div class="label" id="recipient-label"></div>
-                            <div class="address" id="recipient-address"></div>
-                        </div>
-                    </div>
-                </div>
+        /** @type {HTMLElement} */
+        const $recipientLabel = ($recipient.querySelector('.label'));
+        if (request.shopOrigin) {
+            $recipientLabel.textContent = LayoutStandard._originToDomain(request.shopOrigin);
+            $recipientLabel.classList.remove('display-none');
+        } else if (request.recipientLabel) {
+            $recipientLabel.textContent = request.recipientLabel;
+            $recipientLabel.classList.remove('display-none');
+        }
 
-                <div class="nq-card-footer">
-                    <div class="total">
-                        <span id="value"></span><span class="nim-symbol"></span>
-                    </div>
+        super(request, resolve, reject, $recipient);
+    }
 
-                    <div class="fee-section nq-text-s display-none">
-                        <span data-i18n="sign-tx-includes">includes</span>
-                        <span id="fee"></span>
-                        <span class="nim-symbol"></span>
-                        <span data-i18n="sign-tx-fee">fee</span>
-                    </div>
+    /**
+     * @param {string} [origin]
+     * @returns {string}
+     */
+    static _originToDomain(origin) {
+        if (!origin) return '---';
+        return origin.split('://')[1] || '---';
```

Comment:
`shopOrigin` is already parsed when we come to this function, so the tests are unnecessary I think.